### DISCUSSION TITLE
pg_dump version mismatch error during Hasura migration creation

### DISCUSSION THREAD

**Discussion by vlance11** [hasura/graphql-engine#10002] [General]
## Environment
- Hasura CLI version: v2.35.1
- PostgreSQL server version: 16.0 (Debian 16.0-1.pgdg120+1)
- `pg_dump` utility version: 15.4 (Ubuntu 15.4-2.pgdg22.04+1)

## Issue
I'm encountering a version mismatch error between `pg_dump` and my PostgreSQL server when I try to create a migration file using the Hasura CLI. The command I'm using is:

```bash
hasura migrate create "init" --from-server
```

When I select the database to use (default), I get the following error:

```json
{
  "error": "error while executing pg_dump",
  "path": "$",
  "code": "unexpected",
  "internal": "pg_dump: error: aborting because of server version mismatch\npg_dump: detail: server version: 16.0 (Debian 16.0-1.pgdg120+1); pg_dump version: 15.4 (Ubuntu 15.4-2.pgdg22.04+1)\n"
}
```
## Attempts to Resolve

- I've verified the versions of both the server and the `pg_dump` utility.
- I've searched for similar issues in the discussions but didn't find a solution that addresses the version mismatch specifically between PostgreSQL 16.0 and `pg_dump` 15.4.

## Questions

1. Is there a known workaround to resolve this version mismatch?
2. Would updating the Hasura CLI or any other related tooling resolve this issue?
3. Any guidance on how to properly align the `pg_dump` version with the PostgreSQL server version would be appreciated.

Thank you in advance for your support and suggestions!

**Comment by SamirTalwar** [comment]
Hi @vlance11, thanks for reporting this. We need to upgrade `pg_dump` inside the Hasura GraphQL Engine Docker image. This change should be part of the next release.

Are you using Hasura Cloud, Hasura OSS, or another version?

**Comment by danieljharvey** [comment]
The version of `pg_dump` in `graphql-engine` has now been upgraded, this should be resolved from `v2.36.0` onwards.
